uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/leave-request.model.spec.js

```diff
@@ -38,86 +51,92 @@ define([
-        leaveRequestPromise.then(function (response) {
-          expect(response.list.every(function (modelInstance) {
-            return ('cancel' in modelInstance) && ('update' in modelInstance);
-          })).toBe(true);
-        });
+      it('returns according model instances', function () {
+        // It is expected that all types of leave requests are mocked
+        expect(leaveRequestPromiseResult.list.every(function (modelInstance) {
+          return testIfModalInstanceCorrespondsToRequestType(modelInstance);
+        })).toBe(true);
       });
     });
 
     describe('balanceChangeByAbsenceType()', function () {
-      var leaveRequestPromise;
-
       beforeEach(function () {
-        leaveRequestPromise = LeaveRequest.balanceChangeByAbsenceType(jasmine.any(String), jasmine.any(String));
-      });
-
-      afterEach(function () {
-        // to excute the promise force an digest
-        $rootScope.$apply();
+        LeaveRequest.balanceChangeByAbsenceType(1, 2, 3, 4);
+        $rootScope.$digest();
       });
 
       it('calls equivalent API method', function () {
-        leaveRequestPromise.then(function () {
-          expect(LeaveRequestAPI.balanceChangeByAbsenceType).toHaveBeenCalled();
-        });
+        expect(LeaveRequestAPI.balanceChangeByAbsenceType).toHaveBeenCalledWith(1, 2, 3, 4);
       });
     });
 
     describe('find()', function () {
-      var leaveRequestPromise;
+      var leaveRequestPromiseResult;
 
-      beforeEach(function () {
-        leaveRequestPromise = LeaveRequest.find();
-      });
+      describe('basic tests', function () {
+        beforeEach(function () {
+          var requestId = LeaveRequestData.all().values[0].id;
 
-      it('calls equivalent API method', function () {
-        leaveRequestPromise.then(function () {
+          LeaveRequest.find(requestId)
+            .then(function (promiseResult) {
+              leaveRequestPromiseResult = promiseResult;
+            });
+
+          $rootScope.$digest();
+        });
+
+        it('calls equivalent API method', function () {
           expect(LeaveRequestAPI.find).toHaveBeenCalled();
         });
       });
 
-      it('returns model instances', function () {
-        leaveRequestPromise.then(function (response) {
-          expect(('cancel' in response) && ('update' in response)).toBe(true);
+      describe('tests per each request type', function () {
+        ['leave', 'sickness', 'toil'].forEach(function (requestType) {
+          describe('if the type is "' + requestType + '"', function () {
+            beforeEach(function () {
+              var requestId = LeaveRequestData.findBy('request_type', requestType).id;
+
+              LeaveRequest.find(requestId).then(function (promiseResult) {
+                leaveRequestPromiseResult = promiseResult;
+              });
+              $rootScope.$digest();
+            });
+
+            it('returns corresponding model instance', function () {
```

Comment:
it returns a model instance of the corresponding type

Reply:
Done.